libs/runtime/ebpf_ring_buffer.c

```diff
@@ -8,78 +8,279 @@
 
 typedef struct _ebpf_ring_buffer
 {
-    ebpf_lock_t lock;
     size_t length;
-    size_t consumer_offset;
-    size_t producer_offset;
-    uint8_t* shared_buffer;
-    ebpf_ring_descriptor_t* ring_descriptor;
+    volatile size_t consumer_offset;         ///< Consumer has read up to this offset.
+    volatile size_t producer_offset;         ///< Producer(s) have reserved up to this offset.
+    volatile size_t producer_reserve_offset; ///< Next record to be reserved.
+    uint8_t* shared_buffer;                  ///< Shared double-mapped buffer.
+    ebpf_ring_descriptor_t* ring_descriptor; ///< Memory ring descriptor.
 } ebpf_ring_buffer_t;
 
-inline static size_t
-_ring_get_length(_In_ const ebpf_ring_buffer_t* ring)
+static_assert(sizeof(size_t) == 8, "size_t must be 8 bytes");
+
+/**
+ * @brief Read-acquire the record header.
+ *
+ * Used to check for the lock bit to ensure that all writes to the record are visible.
+ *
+ * @param[in] record Pointer to the record.
+ * @return 32 bit record header with length and locked+discard bits.
```

Comment:
```suggestion
 * @return 32 bit record header with length and lock+discard bits.
```
Be consistent in terms of whether the bit is called "lock" or "locked"

Reply:
Thanks. I updated it to consistently use "lock" for the name of the bit that indicates the record is locked.